samples/nrf9160/nrf_cloud_mqtt_multi_service/tests/application/src/main.c

```diff
@@ -0,0 +1,352 @@
+		     call_buf, msg);
+	zassert_true(cJSON_HasObjectItem(msg_obj, NRF_CLOUD_MSG_TIMESTAMP_KEY), "The result of "
+	             "%s must contain a timestamp. Full object: %s",
+		     call_buf, msg);
+
+	/* Grab the property objects */
+	cJSON *appid_obj =	cJSON_GetObjectItem(msg_obj,  NRF_CLOUD_JSON_APPID_KEY);
+	cJSON *msg_type_obj =	cJSON_GetObjectItem(msg_obj,  NRF_CLOUD_JSON_MSG_TYPE_KEY);
+	cJSON *timestamp_obj =	cJSON_GetObjectItem(msg_obj,  NRF_CLOUD_MSG_TIMESTAMP_KEY);
+
+	/* Sanity check. Non-verbose assertion failure message since this should never fail */
+	zassert_not_null(appid_obj,	"appid_obj"	" is null. This should be impossible.");
+	zassert_not_null(msg_type_obj,	"msg_type_obj"	" is null. This should be impossible.");
+	zassert_not_null(timestamp_obj,	"timestamp_obj"	" is null. This should be impossible.");
+
+	/* The expected properties should all be of the correct type */
+	zassert_true(cJSON_IsString(appid_obj),
+		     "The appid of the object returned by %s must be a string. "
+		     "Full object: %s", call_buf, msg);
+	zassert_true(cJSON_IsString(msg_type_obj),
+		     "The message type of the object returned by %s must be a string. "
+		     "Full object: %s", call_buf, msg);
+	zassert_true(cJSON_IsNumber(timestamp_obj),
+		     "The timestamp of the object returned by %s must be a number. "
+		     "Full object: %s", call_buf, msg);
+
+	/* The expected properties should all have the expected value */
+	zassert_true(strcmp(appid_obj->valuestring, sensor) == 0,
+		     "The appid of the object returned by  %s must be \"%s\", "
+		     "but was \"%s\" instead. Full object: %s",
+		     call_buf, sensor, appid_obj->valuestring, msg);
+	zassert_true(strcmp(msg_type_obj->valuestring, NRF_CLOUD_JSON_MSG_TYPE_VAL_DATA) == 0,
+		     "The message type of the object returned by %s must be \"%s\", "
+		     "but was \"%s\" instead. Full object: %s",
+		     call_buf, NRF_CLOUD_JSON_MSG_TYPE_VAL_DATA, msg_type_obj->valuestring, msg);
+
+	snprintf(print_buf, sizeof(print_buf), "%.3f", timestamp_obj->valuedouble);
+	zassert_true(fabs(timestamp_obj->valuedouble - (double)ts) < 0.9,
+		     "The timestamp of the object returned by %s "
+		     "must be %lld, but was %s instead. Full object: %s",
+		     call_buf, ts, print_buf, msg);
+
+	msg_clean(&msg_obj, &msg);
+}
+
+/**
+ * @brief Call create_timestamped_data_message_object repeatedly with a variety of inputs,
+ * making sure that the created objects always meet expectations.
+ */
+static void test_create_timestamped_data_message_object(void) {
+	/* Make sure cJSON is using the correct free/malloc */
+	cJSON_Init();
+
+	/* Generate a variety of messages and make sure they conform to expectations */
+	msg_test(TEST_TIMESTAMP_A, TEST_SENSOR_NAME_A);
+	msg_test(TEST_TIMESTAMP_B, TEST_SENSOR_NAME_B);
+	msg_test(TEST_TIMESTAMP_A, TEST_SENSOR_NAME_B);
+	msg_test(TEST_TIMESTAMP_B, TEST_SENSOR_NAME_A);
+}
+
+static void test_create_timestamped_data_message_object_fail(void) {
+	//TODO
```

Comment:
comment style

Reply:
Oh, yeah, I use `//TODO` specifically so that compliance will scream at me if I forget any TODOs